src/main/java/org/lobid/resources/ElasticsearchIndexer.java

```diff
@@ -469,16 +469,39 @@ public void updateAliases() {
 				groupByIndexCollection(indexName);
 		for (String prefix : indices.keySet()) {
 			final SortedSet<String> indicesForPrefix = indices.get(prefix);
-			final String newIndex = indicesForPrefix.last();
+			final String newestIndex = indicesForPrefix.last();
 			final String newAlias = prefix + aliasSuffix;
-			LOG.info("Prefix " + prefix + ", newest index: " + newIndex);
-			removeOldAliases(indicesForPrefix, newAlias);
-			if (!indexName.equals(newAlias) && !newIndex.equals(newAlias))
-				createNewAlias(newIndex, newAlias);
+			LOG.info("Prefix " + prefix + ", newest index: " + newestIndex);
+			if (!newestIndex.equals(newAlias)) {
+				removeAlias(indicesForPrefix, newAlias);
+				createNewAlias(newestIndex, newAlias);
+			}
 			deleteOldIndices(indexName, indicesForPrefix);
 		}
 	}
 
+	/**
+	 * This adds the productive_prefix as alias to the newest index (which names
+	 * begins with productive_prefix) and adds the staging alias to the second
+	 * newest index name.
+	 * 
+	 * @param productive_prefix the productive alias (and also the prefix of the
+	 *          names of the indices)
+	 * @param staging the "-staging" alias
+	 */
+	public void swapProductionAndStagingAliases(final String productive_prefix,
```

Comment:
To follow the naming conventions of the other variables, this should probably be `productivePrefix` (or, since it's actually the full productive alias, maybe just `productive`).

Otherwise, +1